core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsRaftManager.java

```diff
@@ -0,0 +1,189 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import org.infinispan.commons.io.ByteBuffer;
+import org.infinispan.commons.io.ByteBufferImpl;
+import org.infinispan.commons.util.concurrent.CompletableFutures;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.remoting.transport.raft.RaftChannel;
+import org.infinispan.remoting.transport.raft.RaftChannelConfiguration;
+import org.infinispan.remoting.transport.raft.RaftManager;
+import org.infinispan.remoting.transport.raft.RaftStateMachine;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+import org.jgroups.JChannel;
+import org.jgroups.fork.ForkChannel;
+import org.jgroups.protocols.FORK;
+import org.jgroups.protocols.pbcast.GMS;
+import org.jgroups.protocols.raft.ELECTION;
+import org.jgroups.protocols.raft.InMemoryLog;
+import org.jgroups.protocols.raft.NO_DUPES;
+import org.jgroups.protocols.raft.RAFT;
+import org.jgroups.protocols.raft.REDIRECT;
+import org.jgroups.protocols.raft.StateMachine;
+import org.jgroups.raft.RaftHandle;
+import org.jgroups.stack.ProtocolStack;
+
+/**
+ * A {@link RaftManager} implementation that use JGroups-RAFT protocol.
+ * <p>
+ * This implementation uses multiple {@link ForkChannel}, one of each {@link RaftStateMachine} registered, and it
+ * expects the {@link  FORK} protocol to be present in the main {@link  JChannel}.
+ *
+ * @since 14.0
+ */
+class JGroupsRaftManager implements RaftManager {
+
+   private static final Log log = LogFactory.getLog(JGroupsRaftManager.class);
+
+   private final JChannel mainChannel;
+   private final List<String> raftMembers;
+   private final String raftId;
+   private final Map<String, JgroupsRaftChannel<? extends RaftStateMachine>> raftStateMachineMap = new ConcurrentHashMap<>(16);
+
+   JGroupsRaftManager(GlobalConfiguration globalConfiguration, JChannel mainChannel) {
+      if (mainChannel.getProtocolStack().findProtocol(FORK.class) == null) {
```

Comment:
You have a method `findFork()` in `JGroupsTransport`. You could make it static and reuse that.